pkg/capabilities/utils.go

```diff
@@ -0,0 +1,145 @@
+package capabilities
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+func UnwrapRequest(request CapabilityRequest, config proto.Message, value proto.Message) (bool, error) {
+	migrated, err := FromValueOrAny(request.Inputs, request.Request, value)
+	if err != nil {
+		return migrated, err
+	}
+
+	_, err = FromValueOrAny(request.Config, request.ConfigAny, config)
+	if err != nil {
+		return migrated, err
+	}
+
+	return migrated, nil
+}
+
+func UnwrapResponse(response CapabilityResponse, value proto.Message) (bool, error) {
+	migrated, err := FromValueOrAny(response.Value, response.ResponseValue, value)
+	if err != nil {
+		return migrated, err
+	}
+
+	return migrated, nil
+}
+
+func SetResponse(response *CapabilityResponse, migrated bool, value proto.Message) error {
+	if migrated {
+		wrapped, err := anypb.New(value)
+		if err != nil {
+			return err
+		}
+		response.ResponseValue = wrapped
+		return nil
+	}
+
+	wrapped, err := values.WrapMap(value)
+	if err != nil {
+		return err
+	}
+
+	response.Value = wrapped
+	return nil
+}
+
+func FromValueOrAny(value values.Value, any *anypb.Any, into proto.Message) (bool, error) {
+	if any == nil {
+		if value == nil {
+			return false, errors.New("neither value nor any provided")
+		}
+		err := value.UnwrapTo(into)
+		return false, err
+	}
+
+	err := any.UnmarshalTo(into)
+	return true, err
```

Comment:
_nit_ docstring on func.  what does `true` mean in the case that err != nil?

Reply:
You should generally ignore results when err != nil.

Reply:
agreed, just didn't understand that the `bool` meant `migrated` or not.